tests/Controller/CheckoutShippingApiTest.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Tests\Controller;
+
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class CheckoutShippingApiTest extends CheckoutApiTestCase
+{
+    /**
+     * @test
+     */
+    public function it_denies_order_shipping_selection_for_non_authenticated_user()
+    {
+        $this->client->request('PUT', '/api/checkouts/select-shipping/1');
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'authentication/access_denied_response', Response::HTTP_UNAUTHORIZED);
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_to_select_shipping_for_unexisting_order()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+
+        $this->client->request('PUT', '/api/checkouts/select-shipping/1', [], [], static::$authorizedHeaderWithContentType);
+
+        $response = $this->client->getResponse();
+        $this->assertResponseCode($response, Response::HTTP_NOT_FOUND);
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_to_select_shipping_for_order_that_is_not_addressed()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $checkoutData = $this->loadFixturesFromFile('resources/checkout.yml');
+
+        $url = sprintf('/api/checkouts/select-shipping/%d', $checkoutData['order1']->getId());
+        $this->client->request('PUT', $url, [], [], static::$authorizedHeaderWithContentType);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'checkout/shipping_invalid_order_state', Response::HTTP_INTERNAL_SERVER_ERROR);
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_to_select_shipping_for_order_without_specifying_shipping_method()
+    {
+        // TO-DO
+        // feature cannot be tested properly due to bug with cascade shipments validation
```

Comment:
Can we create a issue with this one? I'm afraid that we will forget for some time about it, if we don't.